state_tracker.go

```diff
@@ -34,13 +34,27 @@ type SerializableState struct {
 	GhostferryVersion         string
 	LastKnownTableSchemaCache TableSchemaCache
 
-	LastSuccessfulPrimaryKeys map[string]uint64
-	CompletedTables           map[string]bool
-	LastWrittenBinlogPosition mysql.Position
+	LastSuccessfulPrimaryKeys                 map[string]uint64
+	CompletedTables                           map[string]bool
+	LastWrittenBinlogPosition                 mysql.Position
+	LastStoredBinlogPositionForInlineVerifier mysql.Position
 }
 
 func (s *SerializableState) MinBinlogPosition() mysql.Position {
-	return s.LastWrittenBinlogPosition
+	nilPosition := mysql.Position{}
+	if s.LastWrittenBinlogPosition == nilPosition {
+		return s.LastStoredBinlogPositionForInlineVerifier
+	}
+
+	if s.LastStoredBinlogPositionForInlineVerifier == nilPosition {
+		return s.LastWrittenBinlogPosition
+	}
+
+	if s.LastWrittenBinlogPosition.Compare(s.LastStoredBinlogPositionForInlineVerifier) >= 0 {
```

Comment:
What exactly is this comparing? We're returning the lesser of the two `LastWrittenBinlogPosition` and `LastStoredBinlogPositionForInlineVerifier`?

Reply:
These two positions can be out of sync because the BinlogWriter independently from adding the binlog events to the BinlogVerifyStore. However, to resume, we need to resume from a single binlog coordinate, which should be the minimum one.